pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -761,6 +764,24 @@ int kprobe__tcp_close(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kretprobe/tcp_close")
+int kretprobe__tcp_close(struct pt_regs* ctx) {
+    u32 cpu = bpf_get_smp_processor_id();
+    batch_t *batch_ptr = bpf_map_lookup_elem(&tcp_close_batch, &cpu);
+    if (batch_ptr == NULL) {
+        return 0;
+    }
+
+    if (batch_ptr->pos == TCP_CLOSED_BATCH_SIZE) {
+        batch_t batch_copy = {};
+        __builtin_memcpy(&batch_copy, batch_ptr, sizeof(batch_copy));
```

Comment:
Let's add a comment explaining why this is necessary since it's easy enough to remove this part and have it work on most kernels.

Reply:
Done.